src/firefly/js/visualize/ChartsContainer.jsx

```diff
@@ -2,24 +2,41 @@
  * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
  */
 
-import React, {Component} from 'react';
+import React, {Component, PropTypes} from 'react';
 import sCompare from 'react-addons-shallow-compare';
+import shallowequal from 'shallowequal';
+
 
 import {flux} from '../Firefly.js';
+import {get} from 'lodash';
 
-import {LO_EXPANDED, dispatchSetLayoutMode, getExpandedMode} from '../core/LayoutCntlr.js';
+import {LO_VIEW, LO_MODE, dispatchSetLayoutMode, getExpandedMode} from '../core/LayoutCntlr.js';
 import {CloseButton} from '../ui/CloseButton.jsx';
 
 import {ChartsTableViewPanel} from '../visualize/ChartsTableViewPanel.jsx';
+import {CHART_SPACE_PATH} from '../visualize/ChartsCntlr.js';
+
+export function getExpandedChartProps() {
+    return get(flux.getState(),[CHART_SPACE_PATH, 'ui', 'expanded']);
+}
+
+function nextState(props) {
+    const {closeable, chartId, tblId, optionsPopup} = props;
+    const expandedMode = props.expandedMode && getExpandedMode() === LO_VIEW.xyPlots;
+    const chartProps = expandedMode ? getExpandedChartProps() : {chartId, tblId, optionsPopup};
+    return Object.assign({expandedMode,closeable}, chartProps);
+}
 
 export class ChartsContainer extends Component {
     constructor(props) {
         super(props);
-        const expandedMode = getExpandedMode() === LO_EXPANDED.xyPlots.view;
-        this.state = {
-            expandedMode: props.expandedMode,
-            closeable: props.closeable
-        };
+        this.state = nextState(this.props);
+    }
+
+    componentWillReceiveProps(np) {
+        if (!this.isUnmounted && !shallowequal(this.props, np)) {
```

Comment:
this.isUnmounted is used here and a few lines below.  however, it was never set.  should be set in componentWillUnmount.